edb/pgsql/delta.py

```diff
@@ -3017,12 +3019,28 @@ def create_index(cls, index, schema, context):
         else:
             sql_exprs = [codegen.SQLSourceGenerator.to_source(sql_tree)]
 
+        except_expr = index.get_except_expr(schema)
+        if except_expr and not except_expr.irast:
+            except_expr = type(except_expr).compiled(
+                except_expr,
+                schema=schema,
+                options=options,
+            )
+        if except_expr:
+            except_tree = compiler.compile_ir_to_sql_tree(
+                except_expr.irast.expr, singleton_mode=True)
+            except_src = codegen.SQLSourceGenerator.to_source(except_tree)
+            except_src = f'({except_src}) is not true'
```

Comment:
We need to be clear in the documentation that if the predicate returns an empty set, we _allow_ the object to be indexed/constrained.